src/ol/CollectionEventType.js

```diff
@@ -9,13 +9,11 @@ export default {
   /**
    * Triggered when an item is added to the collection.
    * @event module:ol/Collection.CollectionEvent#add
-   * @api
    */
   ADD: 'add',
   /**
    * Triggered when an item is removed from the collection.
    * @event module:ol/Collection.CollectionEvent#remove
```

Comment:
Should we add a typedef version of this enum for the apidoc and keep using the enum? Using only typedef for these does not catch typos.

Reply:
I'm envisioning a broader change to handle event types. For the `on`, `once` and `un` API methods, we added type safety already a while ago with separate typedefs - in this case in https://github.com/openlayers/openlayers/blob/main/src/ol/Collection.js#L47-L54.

I started a branch a few day ago to bring this kind of type safety also to the internal methods. I created a draft pull request for this now: #13840.